app/code/core/Mage/Core/Model/Session/Abstract/Varien.php

```diff
@@ -136,33 +137,34 @@ public function start($sessionName=null)
 
         session_start();
 
+        Mage::dispatchEvent('session_before_renew_cookie', ['cookie' => $cookie]);
+
         if (Mage::app()->getFrontController()->getRequest()->isSecure() && empty($cookieParams['secure'])) {
             // secure cookie check to prevent MITM attack
-            $secureCookieName = $sessionName . '_cid';
-            if (isset($_SESSION[self::SECURE_COOKIE_CHECK_KEY])) {
-                if ($_SESSION[self::SECURE_COOKIE_CHECK_KEY] !== md5($cookie->get($secureCookieName))) {
-                    session_regenerate_id(false);
-                    $sessionHosts = $this->getSessionHosts();
-                    $currentCookieDomain = $cookie->getDomain();
-                    foreach (array_keys($sessionHosts) as $host) {
-                        // Delete cookies with the same name for parent domains
-                        if (strpos($currentCookieDomain, $host) > 0) {
-                            $cookie->delete($this->getSessionName(), null, $host);
-                        }
+            $secureCookieName = session_name() . '_cid';
```

Comment:
$sessionName is passed anyway, which switching to method?

Reply:
Because $sessionName can be null. I suppose it could be `$this->getSessionName()` but it would be functionally equivalent.